File: 0078-Subsets/lc0078.cc
// C++ solution.

class Solution {
public:
  vector<vector<int>> subsets(vector<int>& nums) {
    int n = 1 << nums.size();
    vector<vector<int>> ans(n, vector<int>());
    for (int i = 0; i < n; i++) {
      for (int p = nums.size()-1; p >= 0; p--) {
        if ((i >> p) & 1) {
          ans[i].push_back(nums[p]);
        }
      }
    }
    return ans;
  }
};

File: 0078-Subsets/lc0078.rs
// Rust solution.

impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let n: usize = 1 << nums.len();
        let mut ans: Vec<Vec<i32>> = vec![vec![]; n];
        for i in 0..n {
            for p in 0..nums.len() {
                if (i >> p) & 1 == 1 {
                    ans[i].push(nums[p]);
                }
            }
        }
        ans
    }
}
